src/services/navigate.ts

```diff
@@ -0,0 +1,122 @@
+import { ServiceScene } from '../Components/ServiceScene/ServiceScene';
+import { IndexScene } from '../Components/IndexScene/IndexScene';
+import { ALL_VARIABLE_VALUE } from './variables';
+import { getMetadataService } from './metadata';
+import { locationService } from '@grafana/runtime';
+import {
+  buildServicesUrl,
+  DRILLDOWN_URL_KEYS,
+  PageSlugs,
+  prefixRoute,
+  replaceSlash,
+  ROUTES,
+  ValueSlugs,
+} from './routing';
+import { sceneGraph } from '@grafana/scenes';
+import { UrlQueryMap, urlUtil } from '@grafana/data';
+
```

Comment:
I think it would be positive to cover these functions with a few test cases to ensure `locationService` is called with the expected routes. It should also surface any hidden bug.

Reply:
Yeah good call, I also added some test cases to ensure the state in the metadata service is populated as well